File: Safari-611.1.10/LayoutTests/http/tests/misc/webtiming-slow-load.php
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
<html>
<head>
<script>
</script>
<script src="../../js-test-resources/js-test-pre.js"></script>
</head>
<body>
<p id="description"></p>
<div id="console"></div>
<script>
description("Verifies that requestStart and responseStart are available before the main document has finished loading.");

window.performance = window.performance || {};
var navigation = performance.navigation || {};
var timing = performance.timing || {};

shouldBeNonZero("timing.requestStart");
shouldBeNonZero("timing.responseStart");
shouldBe("timing.responseEnd", "0");

</script>
<script src="../../js-test-resources/js-test-post.js"></script>
<?php
    echo str_repeat(" ", 100000);
    flush();
    ob_flush();
    sleep(1);
?>
</body>
</html>

File: LayoutTests/http/tests/misc/webtiming-slow-load.py
#!/usr/bin/env python3

import sys
import time

sys.stdout.write(
    'Content-Type: text/html\r\n\r\n'
    '<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">\n'
    '<html>\n'
    '<head>\n'
    '<script>\n'
    '</script>\n'
    '<script src="../../js-test-resources/js-test-pre.js"></script>\n'
    '</head>\n'
    '<body>\n'
    '<p id="description"></p>\n'
    '<div id="console"></div>\n'
    '<script>\n'
    'description("Verifies that requestStart and responseStart are available before the main document has finished loading.");\n'
    '\n'
    'window.performance = window.performance || {};\n'
    'var navigation = performance.navigation || {};\n'
    'var timing = performance.timing || {};\n'
    '\n'
    'shouldBeNonZero("timing.requestStart");\n'
    'shouldBeNonZero("timing.responseStart");\n'
    'shouldBeNonZero("timing.responseEnd");\n'
    '\n'
    '</script>\n'
    '<script src="../../js-test-resources/js-test-post.js"></script>\n'
)

sys.stdout.write(' ' * 100000)
sys.stdout.flush()
time.sleep(1)

sys.stdout.write(
    '</body>\n'
    '</html>\n'
)
